gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/QueryResultsImpl.java

```diff
@@ -86,8 +89,8 @@ protected T computeNext() {
     if (!entityResultPbIter.hasNext()) {
       return endOfData();
     }
-    DatastoreV1.EntityResult entityResultPb = entityResultPbIter.next();
-    //cursor = entityResultPb.getCursor(); // only available in v1beta3
+    com.google.datastore.v1beta3.EntityResult entityResultPb = entityResultPbIter.next();
+    cursor = entityResultPb.getCursor();
```

Comment:
the comment in EntityResult suggests that the cursor "points to the position after the result entity".
This is why I thought for the last entity in the batch it would be equal to the end_cursor value of the batch (but I think now I can see why it does not have to be the case).

However, I do wonder, if EntityResult.cursor will always have a value even for the last value for the complete query result. I assume the answer is yes.

Also, I assume QueryResultBatch.end_cursor will always have a value even for the last batch for the query.